src/x/parsers/metric_parsers.go

```diff
@@ -0,0 +1,56 @@
+package parsers
+
+import (
+	"bytes"
+
+	"github.com/m3db/m3/src/metrics/metric/id"
+	"github.com/m3db/m3/src/x/serialize"
+)
+
+// GetMetricIDForHistogramAgg returns a metric id (with sorted tag pairs) without the le tag (if is a histogram) and
+// strips histogram suffixes ("_bucket", "_sum". "_count")
+func GetMetricIDForHistogramAgg(metricID id.RawID) (id.RawID, bool) {
+	it := serialize.NewUncheckedMetricTagsIterator(serialize.NewTagSerializationLimits())
+	it.Reset(metricID)
+	leTagName := []byte("le")
+	nameTagName := []byte("__name__")
+
+	bucketSuffix := []byte("_bucket")
+	countSuffix := []byte("_count")
+	sumSuffix := []byte("_sum")
+
+	isHistogram := false
+
+	var idForHistogramAgg []byte
+	for it.Next() {
+		tagName, tagValue := it.Current()
+
+		if bytes.Equal(tagName, nameTagName) {
+			// if the __name__ contains a histogram suffix, then we strip the suffix
+			isHistogram = true
+			switch {
+			case bytes.HasSuffix(tagValue, bucketSuffix):
+				tagValue = tagValue[:len(tagValue)-len(bucketSuffix)]
+			case bytes.HasSuffix(tagValue, countSuffix):
+				tagValue = tagValue[:len(tagValue)-len(countSuffix)]
+			case bytes.HasSuffix(tagValue, sumSuffix):
+				tagValue = tagValue[:len(tagValue)-len(sumSuffix)]
+			default:
+				isHistogram = false
+			}
+
+			if !isHistogram {
+				// if the __name__ does not contain a histogram suffix, then return original metricID
+				// since the metric is not a histogram
+				return metricID, false
+			}
+		} else if bytes.Equal(tagName, leTagName) {
```

Comment:
Why do we still need this if it has the bucket suffix?

Reply:
if it has `_bucket`, it contains `le` as well, so we still need to filter this label out when making a copy of the metric id

Reply:
Oh i see what you mean, we DON'T need to do this check if it has `_sum` or `_count` since these metrics will not have `le`. We can save a comparison

Reply:
just pushed another commit